lib/cn/kvset_split.c

```diff
@@ -341,50 +341,48 @@ kblocks_split(
 
         /* split kblock at split_idx */
         err = kblock_split(&kbd, split_kobj, kblks, hlogs, vused);
-        if (err)
-            return err;
-        assert(kblks[LEFT].n_blks > 0 && kblks[RIGHT].n_blks > 0);
+
+        assert(err || (kblks[LEFT].n_blks > 0 && kblks[RIGHT].n_blks > 0));
 
         /* Append kblks[LEFT] to the left kvset, kblks[RIGHT] to the right kvset, and both
          * kblks[LEFT] and kblks[RIGHT] to the commit list
          */
-        for (uint32_t i = 0; i < kblks[LEFT].n_blks; i++) {
+        for (uint32_t i = 0; i < kblks[LEFT].n_blks && !err; i++) {
             uint64_t blkid = kblks[LEFT].blks[i].bk_blkid;
 
             err = blk_list_append(&blks_left->kblks, blkid);
-            if (!err)
-                err = blk_list_append(result->ks[LEFT].blks_commit, blkid);
+            if (!err) {
+                blks_left->bl_vused += vused[LEFT];
 
-            if (err)
-                return err;
+                err = blk_list_append(result->ks[LEFT].blks_commit, blkid);
+            }
 
-            blks_left->bl_vused += vused[LEFT];
         }
 
-        for (uint32_t i = 0; i < kblks[RIGHT].n_blks; i++) {
+        for (uint32_t i = 0; i < kblks[RIGHT].n_blks && !err; i++) {
             uint64_t blkid = kblks[RIGHT].blks[i].bk_blkid;
 
             err = blk_list_append(&blks_right->kblks, blkid);
-            if (!err)
-                err = blk_list_append(result->ks[RIGHT].blks_commit, blkid);
-
-            if (err)
-                return err;
+            if (!err) {
+                blks_right->bl_vused += vused[RIGHT];
 
-            blks_right->bl_vused += vused[RIGHT];
+                err = blk_list_append(result->ks[RIGHT].blks_commit, blkid);
+            }
         }
 
         /* Add the source kblock to the purge list to be destroyed later */
-        err = blk_list_append(result->blks_purge, kblk->kb_kblk.bk_blkid);
-        if (err)
-            return err;
+        if (!err)
+            err = blk_list_append(result->blks_purge, kblk->kb_kblk.bk_blkid);
+
+        for (int i = LEFT; i <= RIGHT; i++)
```

Comment:
Might be more clear as two separate statements

Reply:
Yeah, I did it this way so it's consistent with the `blk_list_init()` calls above.

Reply:
Could always change those, but it doesn't matter! Merge at will